bootstrapvz/common/tools.py

```diff
@@ -112,9 +112,31 @@ def load_data(path):
 
 
 def config_get(path, config_path):
+	"""Retrieve settings from a json or yaml config file
+	Pass it a linear array of strings, it will recursively search into a nested
+	json or yaml confg file. 
+	"""
 	config = load_data(path)
+	
+	# iterate over a linear list of strings
 	for key in config_path:
-		config = config.get(key)
+		
+		# ensure we have a valid key to search for
+		if key is None:
+			raise Exception("Error: empty key searched for '" + str(key) +
+						"', in the file '" + path + "', try adding it.")
+
+		# attempt to retrieve the string
+		try:
+			config = config.get(key)
+		except AttributeError:
+			raise Exception("Error: cannot find the key '" + str(key) +
+							"', in the file '" + path + "', try adding it.")
+
+		# ensure we have a value to pass back
+		if config is None:
+			raise Exception("Error: the key '" + str(key) +
+						" in the file '" + path + "' is not empty, try defining it.")
```

Comment:
This is the case where the key `is` or `is not` empty? Looks like the condition and the message are stating opposite things.